File: listfragment/src/main/java/com/grantgz/listfragment/BaseFragment.java
package com.grantgz.listfragment;

import android.view.View;
import androidx.fragment.app.Fragment;

public class BaseFragment extends Fragment implements View.OnClickListener  {
    protected String TAG = this.getClass().getSimpleName();
    protected String IDENTIFY = String.valueOf(hashCode());


    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    protected void setOnClickListener(View[] views) {
        for (View v : views) {
            v.setOnClickListener(this);
        }
    }

    protected void setOnClickListener(int[] ids) {
        View view = getView();
        if (view == null)
            return;
        for (int id : ids) {
            view.findViewById(id).setOnClickListener(this);
        }
    }

    @Override
    public void onClick(View v) {

    }

    @Override
    public void onStart() {
        super.onStart();
        if (!isHidden())
            onShow();
    }

    @Override
    public void onStop() {
        super.onStop();
        onHide();
    }

    @Override
    public void onHiddenChanged(boolean hidden) {
        super.onHiddenChanged(hidden);
        if (hidden) {
            onHide();
        } else {
            onShow();
        }
    }

    protected void onShow() {

    }

    protected void onHide() {

    }
}

File: list-fragment/src/main/java/com/adazh/list/base/BaseFragment.kt
package com.adazh.list.base

import android.view.View
import androidx.fragment.app.Fragment

open class BaseFragment : Fragment(), View.OnClickListener {

    val TAG = javaClass.simpleName
    val IDENTIFY = hashCode().toString()

    override fun onDestroy() {
        super.onDestroy()
    }

    protected fun setOnClickListener(views: Array<View>) {
        for (v in views) {
            v.setOnClickListener(this)
        }
    }

    protected fun setOnClickListener(ids: IntArray) {
        val view = view ?: return
        for (id in ids) {
            view.findViewById<View>(id).setOnClickListener(this)
        }
    }

    override fun onClick(v: View) {

    }

    override fun onStart() {
        super.onStart()
        if (!isHidden)
            onShow()
    }

    override fun onStop() {
        super.onStop()
        onHide()
    }

    override fun onHiddenChanged(hidden: Boolean) {
        super.onHiddenChanged(hidden)
        if (hidden) {
            onHide()
        } else {
            onShow()
        }
    }

    protected fun onShow() {

    }

    protected fun onHide() {

    }
}
